ml-agents/mlagents/trainers/torch/model_serialization.py

```diff
@@ -40,6 +40,36 @@ def is_exporting():
         return exporting_to_onnx._local_data._is_exporting
 
 
+class TensorNames:
+    BatchSizePlaceholder = "batch_size"
+    SequenceLengthPlaceholder = "sequence_length"
+    VectorObservationPlaceholder = "vector_observation"
+    RecurrentInPlaceholder = "recurrent_in"
+    recurrentInPlaceholderH = "recurrent_in_h"
+    recurrentInPlaceholderC = "recurrent_in_c"
+    VisualObservationPlaceholderPrefix = "visual_observation_"
+    ObservationPlaceholderPrefix = "obs_"
+    PreviousActionPlaceholder = "prev_action"
+    ActionMaskPlaceholder = "action_masks"
+    RandomNormalEpsilonPlaceholder = "epsilon"
+
+    ValueEstimateOutput = "value_estimate"
+    RecurrentOutput = "recurrent_out"
+    recurrentOutputH = "recurrent_out_h"
+    recurrentOutputC = "recurrent_out_c"
```

Comment:
Are these used anywhere?

Reply:
No, but I think there is a benefit in having symmetry with the C# code that does the same thing. I can remove them if someone disagrees strongly with it.

Reply:
Correct me if I'm wrong but seems like in C# `recurrentOutputH` and `recurrentOutputC` are not used either.

Reply:
They are. They are actually the only ones used. recurrentOutput is not used.

Reply:
TensorGenerator and TensorApplier are both using TensorNames.RecurrentOutput but I don't see where they're using recurrentOutputH? same for recurrent input

Reply:
To clarify, the names used are `recurrent_out_h` and `recurrent_out_c` but these are not grabbed from the TensorNames but directly from Barracuda. But you are right, it does not make sense to keep them explicitly here.

Reply:
Thanks for clarifying, that makes sense to me now.